### PR TITLE
React setup

### PR DESCRIPTION
**Pull request by ismail-syed**
Closes https://github.com/vikrambombhi/SimpleSurvey/issues/25

This PR scaffolds a client-side React application hosted in the `./client` directory. The development and production deployment workflow is documented in the `Readme.md`. 

@vikrambombhi, can we run the production steps documented in the readme in this PR before deployment? 

image

image

### CODE REVIEW

File: client/src/components/NewSurveyForm.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { useState, useCallback } from "react";
+
+import { Form, FormLayout, TextField, Button, Card } from "@shopify/polaris";
+
+export function NewPageForm() {
+  const [id, setId] = useState("");
+  const [title, setTitle] = useState("");
```
Comment (JackyChiu) [inline client/src/components/NewSurveyForm.js]: where is the state coming from?

File: client/src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { SurveyList } from "./SurveyList";
+
+export function App() {
+  const [openModal, setOpenModal] = useState(false);
+  const toggleModalHandler = useCallback(() => {
+    setOpenModal(!openModal);
+  }, [openModal]);
+
+  return (
+    <AppProvider i18n={enTranslations}>
```
Comment (JackyChiu) [inline client/src/components/App.js]: do we really need the translations tho

File: client/src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { useState, useCallback } from "react";
+
+import enTranslations from "@shopify/polaris/locales/en.json";
+import { AppProvider, Page } from "@shopify/polaris";
+import { NewPageForm } from "./NewSurveyForm";
+import { NewSurveyModal } from "./NewSurveyModal";
+import { SurveyList } from "./SurveyList";
+
+export function App() {
```
Comment (JackyChiu) [inline client/src/components/App.js]: how does this connect to the `index.html`?

**Comment by JackyChiu** [comment]
don't you need to add instructions to travis and heroku in order to build and deploy this?

File: client/src/components/NewSurveyForm.js (hunk 1 above)
Comment (ismail-syed) [reply client/src/components/NewSurveyForm.js]: React's `useState` manages this state in memory.

File: client/src/components/App.js (hunk 2 above)
Comment (ismail-syed) [reply client/src/components/App.js]: It's required to get a barebones polaris app, but we don't need to leverage this. It's just boilerplate.

File: client/src/components/App.js (hunk 3 above)
Comment (ismail-syed) [reply client/src/components/App.js]: In `client/src/index.js` we call the following which hooks into the `index.html`

```
ReactDOM.render(<App />, document.getElementById("root"));
```

**Comment by ismail-syed** [comment]
@JackyChiu, Just pushed up changes that allow the react front end to get built during the maven build lifecycle. 

You can test this out using 
```
$ mvn package
$ java -jar target/survey-0.0.1-SNAPSHOT.jar
```

File: client/src/components/NewSurveyForm.js (hunk 1 above)
Comment (JackyChiu) [reply client/src/components/NewSurveyForm.js]: I just don't get how your getting all these variables from the inital state which is empty string

Comment (ismail-syed) [reply client/src/components/NewSurveyForm.js]: You can get see the react docs on [state](https://reactjs.org/docs/hooks-state.html#declaring-a-state-variable) for more details.  See the `What does useState return?` section

**Pull request merged**